portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -72,8 +74,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String email = (String) entity.getProperty("email");
       String message = (String) entity.getProperty("message");
       long timestamp = (long) entity.getProperty("timestamp");
+      String userId =  (String) entity.getProperty("userId");
```

Comment:
The other advantage of using authentication tokens is that someone intercepting their POST requests to your site can't (as easily) imitate another user. If you trust the user's browser to provide their userId, users can trivially replace that field with a user ID of their choice.

But this is fine, for now :)

Reply:
Lets hope no hackers try to mess with my site anytime soon :) 
I'll read up on authentication tokens still, I'm curious how the security aspect of that works